tests/basic_event_manager.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+// A basic, single threaded application that only needs one type of
+// subscriber: `CounterSubscriber`.
+//
+// The application has an `EventManager` and can register multiple subscribers
+// of type `CounterSubscriber`.
+extern crate event_manager;
+extern crate vmm_sys_util;
+
+mod common;
+
+use common::CounterSubscriber;
+use event_manager::{EventManager, SubscriberId, SubscriberOps};
+use std::ops::Drop;
+
+struct App {
+    event_manager: EventManager<CounterSubscriber>,
+    subscribers_id: Vec<SubscriberId>,
+}
+
+impl App {
+    fn new() -> Self {
+        Self {
+            event_manager: EventManager::<CounterSubscriber>::new().unwrap(),
+            subscribers_id: vec![],
+        }
+    }
+
+    fn add_subscriber(&mut self) {
+        let counter_subscriber = CounterSubscriber::new();
+        let id = self.event_manager.add_subscriber(counter_subscriber);
+        self.subscribers_id.push(id);
+    }
+
+    fn run(&mut self) {
+        let _ = self.event_manager.run_with_timeout(100);
+    }
+
+    fn inject_events_for(&mut self, subscriber_index: &[usize]) {
+        for i in subscriber_index {
+            let subscriber = self
+                .event_manager
+                .subscriber_mut(self.subscribers_id[*i])
+                .unwrap();
+            subscriber.trigger_event();
+        }
+    }
+
+    fn clear_events_for(&mut self, subscriber_index: &Vec<usize>) {
```

Comment:
Just curious if there's any special reason why `subscriber_index` is a `Vec` here, and a slice in the previous function? Also, something like `subscriber_indices` might be a more appropriate name.